src/main/java/edu/kis/powp/jobs2d/command/manager/DriverCommandManager.java

```diff
@@ -53,9 +56,18 @@ public synchronized DriverCommand getCurrentCommand() {
 	 */
 	public synchronized void setCurrentCommand(DriverCommand commandList) {
 		this.currentCommand = commandList;
+
+		if(this.currentCommand instanceof ICompoundCommand) {
+			analyzer.analyze((ICompoundCommand) this.currentCommand);
+		}
```

Comment:
reimplement using observer. This code can be in a separate subscriber class registred in changePublisher

Reply:
ok, I wondered about this approach earlier, I will implement it ASAP

Reply:
updated in latest hotfix